MatrixSDK/Room/Polls/PollAggregator.swift

```diff
@@ -153,7 +149,23 @@ public class PollAggregator {
         delegate?.pollAggregatorDidStartLoading(self)
         
         session.aggregations.referenceEvents(forEvent: pollStartEventId, inRoom: room.roomId, from: nil, limit: -1) { [weak self] response in
-            guard let self = self else {
+            guard let self else {
+                return
+            }
+            
+            guard let event = response.originalEvent else {
+                // we didn't found the start event
+                self.delegate?.pollAggregator(self, didFailWithError: PollAggregatorError.invalidPollStartEvent)
+                return
+            }
+            
+            if let pollStartEventContent = MXEventContentPollStart(fromJSON: event.content), pollStartEventContent.answerOptions.count >= Constants.minAnswerOptionCount {
+                pollStartedEvent = event
+                self.pollStartEventContent = pollStartEventContent
+                hasBeenEdited = (event.unsignedData.relations?.replace != nil)
+            } else {
+                // we were not able to decrypt the start event content
+                self.delegate?.pollAggregator(self, didFailWithError: PollAggregatorError.invalidPollStartEvent)
```

Comment:
I see a bit of overlap with the body of `buildPollStartContent`.
I think we can avoid duplication like this.

1. Write a reusable function like this

```swift
    private func tryUpdatePollStartedEvent(with event: MXEvent?) {
        guard
            let event = event,
            let eventContent = MXEventContentPollStart(fromJSON: event.content),
            eventContent.answerOptions.count >= Constants.minAnswerOptionCount
        else {
            delegate?.pollAggregator(self, didFailWithError: PollAggregatorError.invalidPollStartEvent)
            return
        }
        
        pollStartedEvent = event
        pollStartEventContent = eventContent
        hasBeenEdited = (event.unsignedData.relations?.replace != nil)
    }
```

2. Call it here and in `buildPollStartContent`
3. Finally I think we can also delete `throws` from `buildPollStartContent`.

Wdyt?